wpa_supplicant/wpa_cli.c

```diff
@@ -217,31 +227,36 @@ static void wpa_cli_close_connection(void)
 		mon_conn = NULL;
 	}
 }
-
+#endif /* INTERACTIVE */
 
 static void wpa_cli_msg_cb(char *msg, size_t len)
 {
 	printf("%s\n", msg);
 }
 
-
+#ifdef INTERACTIVE
 static int _wpa_ctrl_command(struct wpa_ctrl *ctrl, const char *cmd, int print)
 {
-	char buf[4096];
+	char * buf = NULL;
 	size_t len;
 	int ret;
 
+    buf = os_zalloc(sizeof(u8) * 4096);
+	if (!buf){
+		printf ("Failed to allocate mem for command buf\n");
+		return -1;
+	}
 	if (ctrl_conn == NULL) {
 		printf("Not connected to wpa_supplicant - command dropped.\n");
```

Comment:
Looks like leakage, shouldn't we free the `buf` here? And even if this condition is not hit, where this buffer is going to be freed?